skytemple_randomizer/randomizer/npc.py

```diff
@@ -117,16 +117,16 @@ def _smart_replace_text(self, mapped_actor_names_by_lang):
             # Some place names derived from NPCs are mentioned via [CS:P]...[CR], so we'll replace those as well.
             # Croagunk's Swap Shop is mentioned via [CS:E].
             standard_npc_text = re.compile(
-                r"\[CS:(N|Y|P|E)]((?:\s|.)*?)("
+                r"\[CS:(N|Y|P|E)]([^\[]*)("
```

Comment:
`([^\[]*)` matches any character that's not `[`, including newline characters. The idea is that the regex should stop searching if it encounters either another tag (like `[CR]`) or the end of string:

![image](https://github.com/SkyTemple/skytemple-randomizer/assets/15063274/e783f451-7f2d-43bf-bab7-a9f04c6f3f79)